pkg/installer/installer.go

```diff
@@ -150,6 +153,25 @@ func NewInstaller(config *apiv1.KubectlStorageOSConfig, ensureNamespace bool) (*
 		}
 	}
 
+	if validateKubeVersion {
+		currentVersion, err := pluginutils.GetKubernetesVersion(clientConfig)
+		if err != nil {
+			return installer, err
+		}
+
+		jobName := "storageos-operator-kube-version-" + strconv.FormatInt(time.Now().Unix(), 10)
+		minVersion, err := pluginutils.CreateJobAndFetchResult(clientConfig, jobName, config.Spec.GetNamespace(), version.OperatorLatestSupportedURL(), "cat MIN_KUBE_VERSION")
+		// Version 2.5.0-beta.1 doesn't contains the version file. After 2.5.0 has released error handling needs here.
+		if err == nil && minVersion != "" {
+			supported, err := version.IsSupported(currentVersion.String(), minVersion)
+			if err != nil {
+				return installer, err
+			} else if !supported {
+				return installer, fmt.Errorf("current version of Kubernetes is lower than required minimum version [%s]", minVersion)
+			}
+		}
+	}
```

Comment:
would it make sense for this to be its own function in `cmd/plugin/cli/common.go` and for it to be called by upgrade/install/uninstall commands before entering `NewInstaller` at all?